client/src/pages/DirectoryPage/DirectoryPeople.js

```diff
@@ -16,173 +16,324 @@ import image4 from "../../assets/images/profile-picture-4.png";
-      <Banner background={Banner__Directory}>
-        <h3 style={{ margin: 0 }}>Directory: My Friends</h3>
-        <InputTextField
-          type="text"
-          variation="wide"
-          name="search directory"
-          placeholder="Search Groups"
-        />
-      </Banner>
-      <div className="secondNav">
-        <a className="secondNav">
-          <Button text="My Friends" color="purple" size="long" handleClick={handleClick} />
-        </a>
-        <a className="secondNav" href="./groups">
-          <Button text="My Groups" color="outline" size="long" />
-        </a>
-      </div>{" "}
-      <div className="CommunityPage_SortByButton">
-        <div className="CommunityPage_SortByText">Sort By: Location </div>
-      </div>
-      <div className="CommunityPage_body">
-        <div>
-          <header className="CommunityPage__header">Location:</header>
+export default class DirectoryPeople extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      friends: [],
+      others: [],
+      friendRequests: []
+    };
+  }
+
+
+  componentDidMount() {
+    this.getFriends()
+    friends = this.state.friends
+  }
+
+
+  
+  
+  handleClick = e => {
+    ('/home')
+  };
+  
+  getFriends = () => {
+    fetch(`http://127.0.0.1:8000/profile`, {
+      headers: {
+        Authorization: `Token ${localStorage.getItem("token")}`
+      }
+    })
+      .then(res => res.json())
+      .then(json => {
+        this.makeFriends(json)
+      });
+  }
+  
+  makeFriends = (data) => {
+    for (let i = 0; i < data.length; i++) {
+      if (this.trueFriends(data[i].user)) {
+        let user = data[i]
+        //instead of name just whole user
+        friends.push(user)
+      } else if (data[i].user === this.props.userid) {
+        //do nothing
+      } else if (this.props.requests.length != 0) {
+        for (let j = 0; j < this.props.requests.length; j++) {
+          if (data[i].user === this.props.requests[j]) {
+            let user = data[i]
+            friendRequests.push(user)
+          } else {
+            let user = data[i]
+            others.push(user)
+          }
+        }
+      } else {
+        let user = data[i]
+        others.push(user)
+      }
+    }
+    this.setState({
+      friends: friends,
+      others: others,
+      friendRequests: friendRequests
+    })
+  }
+  
+  showFriends = () => {
+    let cards = []
+    this.friendCards(cards)
+    return cards
+  }
+
+  friendCards = (arr) => {
+    let length  = friends.length
+    for (let i = 0; i < length; i++) {
+      //change to get the ith in the list and that objects name and image
+      arr.push(
+        <div className="CommunityPage__momCard-single">
+        <PersonCard image={this.state.friends[i].image} name={this.state.friends[i].first_name} location="Washington D.C." />
+        <button className="FriendButton" onClick={this.removeFriend}>Remove Friend</button>
         </div>
```

Comment:
All our JSX should be inside the `render` method. This works but it's generally a bad practice